third_party/move/move-model/src/builder/exp_builder.rs

```diff
@@ -4264,14 +4316,29 @@ impl<'env, 'translator, 'module_translator> ExpTranslator<'env, 'translator, 'mo
         // is equivalent.
         let mut bindings = BTreeMap::new();
         let mut args = BTreeMap::new();
-        let field_decls = self
-            .get_field_decls_for_pack_unpack(
-                &struct_entry,
-                &struct_name,
-                &struct_name_loc,
-                variant,
-            )?
-            .clone();
+        let (field_decls, is_positional_constructor) = self.get_field_decls_for_pack_unpack(
+            &struct_entry,
+            &struct_name,
+            &struct_name_loc,
+            variant,
+        )?;
+        let field_decls = field_decls.clone();
+        if is_positional_constructor != expected_positional_constructor {
+            let struct_name_display = struct_name.display(self.env());
```

Comment:
We are ignoring the variant information here (when available), which is part of the constructor.

Reply:
`is_positional_constructor` is computed from the struct and also the variant, if this is what you are asking?